en/guide/message_signing.md

```diff
@@ -1,121 +1,157 @@
+
+> **Note** A system that is accepting incorrectly signed packets should provide a UI or signal to indicate that the connection is *insecure*.
+
+This feature might be useful for finding a lost vehicle with a corrupted secret key (the GCS could choose to still display position information, albeit ideally with a different "untrusted" icon).
+
+
+## Secret Key Management {#secret_key}
+
+A secret key is 32 bytes of binary data that are used to create message signatures that can be verified by other holders of the key.
+The key should be created on one system in the network (often a GCS) and shared to other trusted devices via secure channels.
+All systems on a link will use the *same* key. Different links may use different keys.
+
+The secret key should be stored in persistent storage, and should not be exposed via any publicly accessible communication protocol.
+In particular, the key must not be exposed in MAVLink parameters, MAVLink log files or dataflash log files that may be used for public log analysis.
+
+The method of generating the secret key is implementation dependent. 
+For example, it could be generated by:
+* A user-entered string of up to 64 bytes length that is then run through SHA-256.
+* A random key generator.
+
+The secret key may be shared to other devices using the [SETUP_SIGNING](../messages/common.md#SETUP_SIGNING) message. 
+This should be done over a secure link like USB or wired Ethernet (in theory the secret might be shared over any link if the rules for accepting unsigned packets are met, but it should not be broadcast over a radio interface).
+The receiving system must be set up to process these messages and store the received secret key to the appropriate permanent storage.
+
+Autopilots that don't offer MAVLink over USB might create a module that can set the secret key from a command line interface (e.g. the NSH Shell).
+
+> **Tip** We recommend that all GCS implementations should generate the secret key from a sha256 hash of a user-provided passphrase, and share this with connected systems over a secure link. <!-- really? Random number would be safer, and a warning that it needs to be updated/changed -->
+
+
+<!-- the `SETUP_SIGNING` message should be sent separately to all `system_id`, `component_id` targets in the device’s MAVLink routing table. It should not be sent as a broadcast message. This is to prevent recipients from broadcasting the signature over a radio interface via MAVLink routing in the case where a USB connection is used for setting the key, but the vehicle also has an active telemetry radio.
+-->
+
+## Logging
+
+In order to avoid leaking the secret key used for signing, systems should omit [SETUP_SIGNING](../messages/common.md#SETUP_SIGNING) messages from logs (or replace the secret with 32 0xFF bytes in the logged message).
+
+Similarly, signed packets should have the signature bit cleared and the signature block removed before being put into telemetry log files.
```

Comment:
`should have the signature *incompatibility* bit`

Reply:
Done (added that as link).